packages/html/src/HTMLTooltip.ts

```diff
@@ -0,0 +1,303 @@
+import { HTMLWidget } from "@hpcc-js/common";
+import { scopedLogger, ScopedLogging } from "@hpcc-js/util";
+import { select as d3Select } from "d3-selection";
+
+type Direction = "n" | "s" | "e" | "w" | "ne" | "nw" | "se" | "sw";
+type Position = { x: number, y: number };
+type DirectionalBBox = { [key in Direction]: Position; };
+
+type Rectangle = {top: number, left: number, width: number, height: number};
+export class HTMLTooltip extends HTMLWidget {
+    _targetElement;
+    _tooltipElement;
+    _triggerElement;
+    _arrowElement;
+    _tooltipHTMLCallback = (data?) => "<b>_tooltipHTMLCallback is undefined</b>";
```

Comment:
You could have a better default tooltip (a simple html table of the key values if its an obj for example)?